File: python/03-composition/py/step-01.py
# Step 1: Tile Grid with Random Black Shapes (Processing.py)
# ----------------------------------------------------------
# Create a grid of tiles, each with a random black shape (circle, square, or line)
# For use in Processing IDE, Python mode.
#
# This sketch introduces grid iteration, randomness, and basic shape drawing.
# Each cell in the grid randomly draws a black circle, square, or line (all the same size, no rotation).
#
# TEACHER NOTES:
# - Use this to discuss iteration, randomness, and constraints in art and design.
# - Ask students to predict what will happen if you change the grid size or shape selection logic.
# - Challenge students to remix: add color, change grid size, or add new shapes (triangles, arcs, etc).
# - Discuss how randomness can be used for both art and science, and how constraints shape creativity.
#
# CODE WALKTHROUGH:
# - gridSize: sets the number of rows/columns (12x12 by default).
# - For each cell:
#     1. Randomly pick a shape: 0 = circle, 1 = square, 2 = line.
#     2. Draw the shape centered in the cell, all black.
# - All drawing is done in setup(), so the grid is static (not animated).

gridSize = 12  # Number of rows and columns

def setup():
    """
    Processing.py entry point. Draws a grid of tiles, each with a random black shape (circle, square, or line).
    """
    size(600, 600)
    background(255)
    noStroke()
    cellW = width / gridSize
    cellH = height / gridSize
    for i in range(gridSize):
        for j in range(gridSize):
            x = i * cellW + cellW / 2
            y = j * cellH + cellH / 2
            shapeType = int(random(3))  # 0 = circle, 1 = square, 2 = line
            fill(0)
            if shapeType == 0:
                ellipse(x, y, cellW * 0.7, cellH * 0.7)
            elif shapeType == 1:
                rectMode(CENTER)
                rect(x, y, cellW * 0.7, cellH * 0.7)
            else:
                stroke(0)
                strokeWeight(4)
                line(x - cellW * 0.3, y, x + cellW * 0.3, y)
                noStroke()

File: javascript/03-composition/js/step-01.js
// Step 1: 12x12 Tile Grid Composition
// At each grid point, randomly draw a black circle, square, or line (all the same size, no rotation)

const gridSize = 12; // Number of rows and columns in the grid

function setup() {
  createCanvas(800, 800); // Set up a square canvas
  background(255); // White background
  noLoop(); // Only draw once
  rectMode(CENTER); // Draw rectangles from the center
  noStroke(); // No outline for shapes (except lines)
  fill(0); // Black fill for all shapes

  let cellW = width / gridSize; // Width of each grid cell
  let cellH = height / gridSize; // Height of each grid cell

  // Loop through each grid position
  for (let i = 0; i < gridSize; i++) {
    for (let j = 0; j < gridSize; j++) {
      let x = i * cellW + cellW / 2; // Center x of cell
      let y = j * cellH + cellH / 2; // Center y of cell
      let shapeType = int(random(3)); // Randomly pick 0 (circle), 1 (square), or 2 (line)
      let size = cellW; // All shapes same size as cell width
      if (shapeType === 0) {
        ellipse(x, y, size); // Draw circle
      } else if (shapeType === 1) {
        rect(x, y, size, size); // Draw square
      } else {
        stroke(0); // Black line
        strokeWeight(4); // Thicker line
        line(x - size/2, y, x + size/2, y); // Draw horizontal line
        noStroke(); // Turn off stroke for next shape
      }
    }
  }
}

// All shapes are black for this step (constraint)
